weasel/_util.py

```diff
@@ -171,23 +179,54 @@ def substitute_project_variables(
     return dict(interpolated["project"])
 
 
-def validate_project_version(config: Dict[str, Any]) -> None:
-    """If the project defines a compatible spaCy version range, chec that it's
-    compatible with the current version of spaCy.
+def validate_weasel_version(config: Dict[str, Any]) -> None:
+    """If the project defines a compatible Weasel version range, check that it's
+    compatible with the current version of Weasel.
 
     config (Dict[str, Any]): The loaded config.
     """
-    spacy_version = config.get("spacy_version", None)
-    if spacy_version and not is_compatible_version(about.__version__, spacy_version):
+    version_specifier = config.get("weasel_version", None)
+    weasel_version = version("weasel")
+    if version_specifier and not is_compatible_version(
+        weasel_version, version_specifier
+    ):
         err = (
-            f"The {PROJECT_FILE} specifies a spaCy version range ({spacy_version}) "
-            f"that's not compatible with the version of spaCy you're running "
-            f"({about.__version__}). You can edit version requirement in the "
+            f"The {PROJECT_FILE} specifies a Weasel version range ({version_specifier}) "
+            f"that's not compatible with the version of Weasel you're running "
+            f"({weasel_version}). You can edit version requirement in the "
             f"{PROJECT_FILE} to load it, but the project may not run as expected."
         )
         msg.fail(err, exits=1)
 
 
+def validate_spacy_version(config: Dict[str, Any]) -> None:
+    """If the project defines a compatible spaCy version range, check that it's
+    compatible with the current version of spaCy.
+
+    config (Dict[str, Any]): The loaded config.
+    """
+    version_specifier = config.get("spacy_version", None)
+    if version_specifier:
+        try:
+            spacy_version = version("spacy")
+        except PackageNotFoundError:
+            err = (
+                f"The {PROJECT_FILE} specifies a spaCy version range ({version_specifier}) "
+                f"but spaCy is not installed within the environment. "
+                f"Did you forget to add 'spacy{version_specifier}` to your requirements?"
+            )
+            msg.fail(err, exits=1)
```

Comment:
I wouldn't error on this, but instead print a warning. I would also remove the last sentence "Did you forget..." because it proposes to add spaCy to the requirements which might not actually be needed - maybe they just need to specify `weasel`. Like - there might be current projects that don't actually use spaCy but still have a `spacy_version` to pin to the right `projects` functionality - they could instead start specifying the `weasel_version`.

Reply:
If we warn we can set `spacy_version` to `None` and/or avoid running the `if not is_compatible_version` check below.

Reply:
That's fine by me.